src/quick-lint-js/parse.h

```diff
@@ -2515,8 +2517,43 @@ class parser {
     }
 
     if (this->peek().type == token_type::kw_else) {
+      source_code_span else_span = this->peek().span();
       this->skip();
-      parse_and_visit_body();
+
+      switch (this->peek().type) {
+      default:
+        parse_and_visit_body();
+        break;
+
+      case token_type::left_paren:
+        expression *ast = this->parse_expression(precedence{});
+        this->visit_expression(ast, v, variable_context::rhs);
+
+        bool is_invalidating_if = false;
+
+        switch (ast->kind()) {
+        default:
+          break;
+
+        case expression_kind::arrow_function_with_expression:
+        case expression_kind::arrow_function_with_statements:
+          is_invalidating_if = true;
+          break;
+        }
+
+        this->consume_semicolon();
```

Comment:
Blocking: This looks like it's in the wrong spot. Won't skipping a semicolon here cause the following code to have the error_else_with_conditional_missing_if warning?
```js
if (a) { b; } else (c);
{ d; }
```

I think we need to consume a semicolon after the check on line 2546.

Reply:
Agreed!

Reply:
Please add a test for this example.